app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/ResponseBodyRow/index.js

```diff
@@ -52,29 +54,42 @@ const ResponseBodyRow = ({ rowIndex, pair, pairIndex, helperFunctions, ruleDetai
         setIsCodeMinified(true);
         setEditorStaticValue(value);
       }
-
-      modifyPairAtGivenPath(null, pairIndex, `response.type`, responseBodyType, [
-        {
-          path: `response.value`,
-          value: value,
-        },
-        {
-          path: `response.serveWithoutRequest`,
-          value: undefined,
-        },
-      ]);
+      dispatch(
+        actions.updateRulePairAtGivenPath({
+          pairIndex,
+          objectPath: "response.type",
+          newValue: responseBodyType,
+          arrayOfOtherValuesToModify: [
```

Comment:
If we merge the primary update and side-effects in a single object, it will look much cleaner.

```js
actions.updateRulePairAtGivenPath({
  pairIndex,
  updates: {
    "response.type": responseBodyType,
    "response.value": value,
    "response.serveWithoutRequest": undefined
  }
});
```